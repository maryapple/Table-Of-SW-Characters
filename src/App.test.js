import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import App from './app'
import SearchPanel from './search';
import Table from './table';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    // ReactDOM.unmountComponentAtNode(div);
});

it('search renders correctly', () => {
    const search = renderer.create(<SearchPanel  />)
    const tree = search.toJSON();
    expect(tree).toMatchSnapshot();
});

it('table renders correctly', () => {
    const table = renderer
        .create(<Table data={
            [
                {
                "name": "Luke Skywalker",
                "height": "172",
                "mass": "77",
                "hair_color": "blond",
                "skin_color": "fair",
                "eye_color": "blue",
                "birth_year": "19BBY"
                }
            ]
        } />)
    const tree = table.toJSON();
    expect(tree).toMatchSnapshot();
});

it('changes the data in the input when user writes something ', () => {
    const component = renderer.create(<SearchPanel />)
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
