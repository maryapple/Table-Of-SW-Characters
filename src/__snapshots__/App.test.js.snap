// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the data in the input when user writes something  1`] = `
<div>
  <input
    className="form-control col-sm-2"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value=""
  />
</div>
`;

exports[`search renders correctly 1`] = `
<div>
  <input
    className="form-control col-sm-2"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value=""
  />
</div>
`;

exports[`table renders correctly 1`] = `
<table
  className="table table-striped"
>
  <thead
    className="thead-dark"
  >
    <tr>
      <th
        scope="col"
      >
        name
      </th>
      <th
        scope="col"
      >
        height
      </th>
      <th
        scope="col"
      >
        mass
      </th>
      <th
        scope="col"
      >
        hair_color
      </th>
      <th
        scope="col"
      >
        skin_color
      </th>
      <th
        scope="col"
      >
        eye_color
      </th>
      <th
        scope="col"
      >
        birth_year
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      style={
        Object {
          "backgroundColor": "",
        }
      }
    >
      <td>
        Luke Skywalker
      </td>
      <td>
        172
      </td>
      <td>
        77
      </td>
      <td>
        blond
      </td>
      <td>
        fair
      </td>
      <td>
        blue
      </td>
      <td>
        19BBY
      </td>
    </tr>
  </tbody>
</table>
`;
